tests/e2e/validatecluster/validatecluster_test.go

```diff
@@ -168,6 +168,53 @@ var _ = Describe("Verify Basic Cluster Creation", Ordered, func() {
 		}, "120s", "2s").Should(ContainSubstring("kubernetes.default.svc.cluster.local"))
 	})
 
+	It("Verifies Restart", func() {
+		_, err := e2e.DeployWorkload("daemonset.yaml", kubeConfigFile)
+		Expect(err).NotTo(HaveOccurred(), "Daemonset manifest not deployed")
+		defer e2e.DeleteWorkload("daemonset.yaml", kubeConfigFile)
+		nodes, _ := e2e.ParseNodes(kubeConfigFile, false)
+		errRestart := e2e.RestartCluster(serverNodeNames)
+		Expect(errRestart).NotTo(HaveOccurred(), "Restart Nodes not happened correctly")
+		if len(agentNodeNames) > 0 {
+			errRestartAgent := e2e.RestartAgent(agentNodeNames)
+			Expect(errRestartAgent).NotTo(HaveOccurred(), "Restart Agent not happened correctly")
+		}
```

Comment:
I probably made this confusing on my last review, but what I meant was that the first `Eventually` block should run (i.e it checks the condition before restarting), then this restart stuff should happen, then the second `Eventually` block should run (i.e it checks the condition after restarting)